plugins/org.python.pydev.debug/src/org/python/pydev/debug/ui/launching/PythonRunnerConfig.java

```diff
@@ -439,10 +444,19 @@ public PythonRunnerConfig(ILaunchConfiguration conf, String mode, String run,
 
             }
 
+            if (isDebug) {
+                updateVar(pythonNature, manager, win32, envMap, "PATHS_FROM_ECLIPSE_TO_PYTHON",
+                        pathMappings.toString());
+            }
+
             //And we also must get the environment variables specified in the interpreter manager.
             envp = interpreterLocation.updateEnv(envp, envMap.keySet());
         }
 
+        if (isDebug) {
+            envp = StringUtils.addString(envp, "PATHS_FROM_ECLIPSE_TO_PYTHON=" + pathMappings.toString());
+        }
```

Comment:
I don't really know how to use the parsed JSON to create the environment variable. This variable addition here is a draft

Reply:
This code is correct, the only thing is that it should check that the `pathMappings` are valid prior to setting it (if it's not defined we don't want to override what may be set in the environment already).